src/features/broker-protection/comparisons/is-same-name.js

```diff
@@ -133,11 +143,58 @@ export function isSameName (fullNameExtracted, userFirstName, userMiddleName, us
     return false
 }
 
+/**
+ * Given the user's provided name, look for nicknames or full names and return a list
+ *
+ * @param {string | null} name
+ * @return {Set.<string>|Set<*>}
+ */
+export function getNames (name) {
+    if (name == null || name.trim() === '') { return new Set() }
+
+    name = name.toLowerCase()
+
+    return new Set([name, ...getNicknames(name), ...getFullNames(name)])
+}
+
+/**
+ * Given a full name, get a list of nicknames, e.g. Gregory -> Greg
+ *
+ * @param {string | null} name
+ * @return {Set.<string>|Set<*>}
+ */
 export function getNicknames (name) {
-    if (name == null || name.trim() === '') { return [] }
+    const emptySet = new Set()
+
+    if (name == null || name.trim() === '') { return emptySet }
```

Comment:
this same comparison keeps popping up, let's add a helper for it

```js
/**
 * This will handle all none-string types like null / undefined too
 * @param {any} [input]
 * @return {input is string}
 */
function noneEmptyString (input) {
    if (typeof input !== 'string') return false
    return input.trim().length > 0
}
```

then for each place:

```js
if (!noneEmptyString(name)) return new Set()
```

etc..